### PR TITLE
[AIRFLOW-5561] Relax httplib2 version required for gcp extra

### PR DESCRIPTION
**Pull request by camilomartinez**
### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-5561

### Description

As part of [AIRFLOW-3971] a minor version range dependency was
introduced that means that only older versions of httplib2 can
be used.

I'm relaxing this dependency as I tested with the latest 0.13.1
and it works correctly

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Comment by potiuk** [comment]
@mik-laj -> was there any reason why we fixed the dependency to the httplib library ? Any problems with Google client libs with newer version?

**Comment by potiuk** [comment]
I noticed that it was requested by @kaxil (see the conversation in https://issues.apache.org/jira/browse/AIRFLOW-3971 ) 
```
@kaxil 
Can we update this to httplib2~=0.9.2, so that the major version update doesn't break Airflow.

@mik-laj
Yes. I will do it.
```

@kaxil @mik-laj - I am very interested in reviewing the dependencies in general and applying the process of managing them (this is my next area of focus to improve our dev environment). GCP deps are one of the most complex I think because of all the different librarie we use. 

I would love your comments here - whether there were any serious dependencies here or reasons why we've pinned it here.

**Review by kaxil — APPROVED** [review]
LGTM. We pinned this earlier as we have had horrible experiences with major upgrades.

File: setup.py [hunk 1]
```diff
@@ -210,7 +210,7 @@ def write_version(filename: str = os.path.join(*["airflow", "git_version"])):
     'google-cloud-videointelligence>=1.7.0',
     'google-cloud-vision>=0.35.2',
     'grpcio-gcp>=0.2.2',
-    'httplib2~=0.9.2',
+    'httplib2>=0.9.2',
```
Comment (kaxil) [inline setup.py]: Can we instead have `httplib2~=0.9`

**Review by kaxil — CHANGES_REQUESTED** [review]

File: setup.py (hunk 1 above)
Comment (kaxil) [reply setup.py]: This should mean `>= 0.9, == 0.*`

Comment (camilomartinez) [reply setup.py]: Makes sense to me

**Comment by camilomartinez** [comment]
@kaxil I applied the suggested change

**Comment by kaxil** [comment]
Thanks @camilomartinez

**Pull request merged**